File: Cuke4Duke/trunk/NoChangeParking/src/main/java/com/sqs/nochangeparking/core/Payment.java
package com.sqs.nochangeparking.core;

public class Payment {
	
	private String vehicle;
	
	public Payment(String vehicle) {
		this.vehicle = vehicle;
	}
	
	public String getVehicle() {
		return vehicle;
	}
	
	public boolean equals(Object o) {
		if (!(o instanceof Payment)) {
			return false;
		}
		Payment p = (Payment)o;
		return (this.vehicle.equals(p.vehicle));
	}

}

File: DotNetFitNesse/trunk/NoChangeParking/src/NoChangeParking.Core/Payment.cs
using System;

namespace NoChangeParking.Core
{
	public class Payment
	{
		private string vehicle;
		private Money amount;
		
		internal Payment(string vehicle, Money amount) 
		{
			this.vehicle = vehicle;
			this.amount = amount;
		}
		
		public string PaidForVehicle 
		{
			get { return vehicle; }
		}

		public override bool Equals(object obj) 
		{
			return (obj.GetType() == typeof(Payment)) &&
				   ((Payment)obj).vehicle == this.vehicle;
		}

		public override int GetHashCode() 
		{
			return vehicle.GetHashCode();
		}

		public Money Amount {
			get { return amount; }
		}
	}
}
